File: algorithm/sorting.bubble.py
def bubbleSort(nums):
    for i in range(len(nums)-1):
        for j in range(i, len(nums)):
            if nums[j] < nums[i]:
                nums[j], nums[i] = nums[i], nums[j]
    return nums


given = [5, 7, 8, 1, 4, 6, 9]
expected = [1, 4, 5, 6, 7, 8, 9]

output = bubbleSort(given)
assert output == expected

File: algorithm/sorting.bubble.mjs
import assert from "node:assert";

function sort(arr) {
  for (let i = 0; i < arr.length - 1; i += 1) {
    for (let j = i; j < arr.length; j += 1) {
      if (arr[j] < arr[i]) {
        let tmp = arr[j];
        arr[j] = arr[i];
        arr[i] = tmp;
      }
    }
  }
  return arr;
}
let given = [3, 2, 1, 5, 4];
let expected = [1, 2, 3, 4, 5];
let output = sort(given);

assert.deepEqual(output, expected);
// console.assert(output === expected, "Wrong answer");
